system-addon/lib/HighlightsFeed.jsm

```diff
@@ -260,6 +274,18 @@ this.HighlightsFeed = class HighlightsFeed {
       case at.UNINIT:
         this.uninit();
         break;
+
+      // Relay the downloads actions to DownloadsManager
+      case at.NEW_TAB_LOAD:
```

Comment:
You don't need to do this, you can just call `this.downloadsManager.onAction(action);` on every action